### ISSUE TITLE
Document Origins and cluster limitations

### ISSUE THREAD

**Issue by gcalacoci** [2ndQuadrant/pglogical#76]
Right now is not possible for pglogical to support a database that is both a subscriber and a provider within the same node. 
Something like tableA:databaseA --> databaseB and tableB:databaseB --> databaseC

Trying to handle a node with a "dual" role leads pglogical to crash, and recovering from the crash is difficult.

This happens because replication origins are cluster-wide in PostgreSQL, and is an actual pglogical limitation.

Please improve the documentation about origins (that are not explained) and actual technical limitations of pglogical.

Would also be useful to improve the documentation about recovering from a crash.

**Comment by PJMODOS** [comment]
I'd rather look into fixing the limitation first

**Comment by john5000** [comment]
Hi @gcalacoci
Good point on the documentation improvements requested.

But as to using database node as subscriber and provider I'm doing that without problems.
That node subscribes and provides to a separate server which is also subscriber and provider.
That is both nodes subscribe to each other.
(And for testing I've set up triggers to move data between the tables to create a continuous replication loop.  Millions of records pumped through it.)

So am not having trouble with that. 
Using:
pglogical 1.2.1 (using an earlier version I did have trouble. And recovery troubles.)
Postgres 9.5

**Comment by PJMODOS** [comment]
Yes, the problem is only when you create cascaded setup within single postgres instance.
